File: decompiled/3.18.0/be/ehealth/technicalconnector/validator/TimeStampValidator.java
package be.ehealth.technicalconnector.validator;

import be.ehealth.technicalconnector.exception.InvalidTimeStampException;
import be.ehealth.technicalconnector.exception.TechnicalConnectorException;
import java.security.KeyStore;
import java.util.List;
import org.bouncycastle.tsp.TimeStampToken;

public interface TimeStampValidator {
   /** @deprecated */
   @Deprecated
   String ALIASLIST = "timestampvalidatior.aliaslist";
   String KEYSTORE = "timestampvalidatior.keystore";

   void validateTimeStampToken(TimeStampToken var1) throws InvalidTimeStampException, TechnicalConnectorException;

   void validateTimeStampToken(byte[] var1, TimeStampToken var2) throws InvalidTimeStampException, TechnicalConnectorException;

   /** @deprecated */
   @Deprecated
   void setKeyStore(KeyStore var1);

   /** @deprecated */
   @Deprecated
   void setAliases(List<String> var1);
}

File: src/main/kotlin/org/taktik/connector/technical/validator/TimeStampValidator.kt
package org.taktik.connector.technical.validator

import org.taktik.connector.technical.exception.InvalidTimeStampException
import org.taktik.connector.technical.exception.TechnicalConnectorException
import java.security.KeyStore
import org.bouncycastle.tsp.TimeStampToken

interface TimeStampValidator {

    @Throws(InvalidTimeStampException::class, TechnicalConnectorException::class)
    fun validateTimeStampToken(var1: TimeStampToken)

    @Throws(InvalidTimeStampException::class, TechnicalConnectorException::class)
    fun validateTimeStampToken(var1: ByteArray, var2: TimeStampToken)


    @Deprecated("")
    fun setKeyStore(var1: KeyStore?)


    @Deprecated("")
    fun setAliases(var1: MutableList<String>)
}
